drivers/serial/uart_bm.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016 Open-RnD Sp. z o.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for UART port on Bristlemouth
+ *
+ */
+
+#ifndef ZEPHYR_DRIVERS_SERIAL_UART_BM_H_
+#define ZEPHYR_DRIVERS_SERIAL_UART_BM_H_
+
+#include <drivers/pinctrl.h>
+
+#include <stm32_ll_usart.h>
+
+/* device config */
+struct uart_bm_config {
+	/* USART instance */
+	USART_TypeDef *usart;
+	/* clock subsystem driving this peripheral */
+	struct stm32_pclken pclken;
+	/* initial hardware flow control, 1 for RTS/CTS */
+	bool hw_flow_control;
```

Comment:
Given that we have only two wires to communicate over, I'm guessing that we can probably cut out hardware flow control support directly.